app/src/main/java/com/concordia/mcga/adapters/BuildingInformationArrayAdapter.java

```diff
@@ -0,0 +1,212 @@
+package com.concordia.mcga.adapters;
+
+import android.content.Context;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.concordia.mcga.activities.R;
+
+import java.util.ArrayList;
+
+import static android.content.ContentValues.TAG;
+import static com.concordia.mcga.activities.R.id.imageView;
+
+
+public class BuildingInformationArrayAdapter extends ArrayAdapter<String[]> {
+
+    ////////////////////////////////////////////////////////////
+    // INSTANCE VARIABLES
+    ////////////////////////////////////////////////////////////
+
+    private final int IMAGES_PER_ROW = 4;
+
+    private final Context context;
+    private ArrayList<String[]> rowImages = new ArrayList<String[]>();
+
+    // 4 images per per row
+    private String[] image = new String[IMAGES_PER_ROW];
+    private ImageView[] imageView = new ImageView[IMAGES_PER_ROW];
+
+
+
+    ////////////////////////////////////////////////////////////
+    // CLASS METHODS
+    ////////////////////////////////////////////////////////////
+
+    /**
+     * Constructor
+     * @param context application context
+     * @param rowImages Contains the list of images to display
+     */
+    public BuildingInformationArrayAdapter(Context context, ArrayList<String[]> rowImages) {
+        super(context, R.layout.building_info_list, rowImages);
+        this.context = context;
```

Comment:
Did you check the implementation of super(...) in the hierarchy? I did and it does indeed assign `this.context = context`